domains/game/GameScreen.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect } from "react";
+import { ActivityIndicator, View, Text } from "react-native";
+import CONSTANTS from "../../const";
+import PropTypes from "prop-types";
+import Game from "./Game.jsx";
+import Waiting from "./Waiting.jsx";
+
+const isGameFull = (game) => {
+    for (const player of game.players) {
+        if (!player.name) {
+            return false;
+        }
+    }
+    return true;
+};
+
+const GameScreen = ({ route }) => {
+    const { playerName, gameId } = route.params;
+    const [gameState, setGameState] = useState("pending");
+    const [error, setError] = useState();
+    const [game, setGame] = useState();
+
+    useEffect(() => {
+        if (gameState === "pending") {
+            setGameState("loading");
+            fetch(CONSTANTS.BASE_URL + "/game/" + gameId)
+                .then(async (res) => {
+                    const data = await res.json();
+                    if (!res.ok) {
+                        const error = (data && data.message) || res.status;
+                        return Promise.reject(error);
+                    }
+                    setGame(data);
+                    setGameState("loaded");
+                })
+                .catch((err) => {
+                    setError(err);
+                    setGameState("error");
+                });
+        }
+    }, [gameState, setGameState, setError, setGame]);
+    useEffect(() => {
+        const updateGame = setInterval(function () {
+            if (gameState === "pending") {
+                fetch(CONSTANTS.BASE_URL + "/game/" + gameId)
+                    .then(async (res) => {
+                        const data = await res.json();
+                        if (!res.ok) {
+                            const error = (data && data.message) || res.status;
```

Comment:
Error is not an error object it's a string so you will get back a string in .catch and a string in the error state